### DISCUSSION TITLE
MariaDB postinstall step no longer runs in CI environment

### DISCUSSION THREAD

**Discussion by acj** [Homebrew/discussions#287] [Everyday usage] [answered]
I have a CI use case that requires a working MariaDB 10.4 install. As of commit https://github.com/Homebrew/homebrew-core/commit/77e4e051812ad4bc9688e26ef9d346f8aaa78455 (from https://github.com/Homebrew/homebrew-core/pull/66727), the postinstall step isn't run when the `$CI` variable is set, which is fairly common across different build farms (including GitHub Actions runners).

MariaDB can't start, and so I get errors like this:

> ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)

~I can work around this by temporarily unsetting `$CI` and then running `brew postinstall ...` manually, but that feels dirty.~ (See update below)

Is there a better option? Could the postinstall step use a variable that's unique to Homebrew's CI environment?

**Comment by jonchang** [comment, marked as answer]
Probably the best solution is to just run `CI= brew postinstall mariadb` (or even: `CI= brew install mariadb`). As the pull request that added that commit states, on CI boxes basically anything can be installed so skipping `postinstall` on CI makes sense since this caused a lot more issues than enabling postinstall on CI has solved.

**Reply by acj** [reply]
Thanks. After more testing, I found that clearing or unsetting `$CI` when explicitly running `postinstall` doesn't fix the problem. The post-install code runs into the same "can't connect" error that my CI script was. Not sure why.

This does work, though:

```
CI= brew bundle ...
```

I haven't tried `CI= brew install mariadb`, but I imagine it would work as well.

**Reply by gromgit** [reply]
Note that the `mariadb` post-install step only sets up an empty DB if it thinks one doesn't exist. It specifically does _not_ start the DB server, and `mysql_install_db` seems to be especially sensitive to any unexpected stuff it may find already in the designated directory, so if it's failing to work correctly, I don't see why the DB server _wouldn't_ crash on startup too. That's why I created that PR.

For CI purposes, I recommend doing what the `test` block does: create a fresh DB dir in a temp dir.

**Reply by acj** [reply]
> so if it's failing to work correctly, I don't see why the DB server wouldn't crash on startup too

I agree with the logic there (and in the PR). The still-confusing thing is that it was working fine (for my admittedly narrow use case) prior to the linked commit. My CI runners start out in a clean state, and I've confirmed that the designated directory doesn't exist, so it should be safe to run `mysql_install_db`. There seems to be an interaction between the steps that's causing this error.

Anyway, I think I was lucky that this was working prior to the formula change and wasn't aware that my CI install was sensitive to the order of operations. I'll need to rework a couple of things to fit the new model, which is fine. Thanks again for the suggestions.
